src/SyntaxHighlighter/Cpp.cpp

```diff
@@ -10,31 +10,34 @@
 #include <QTextCharFormat>
 
 namespace orbit_syntax_highlighter {
-namespace C {
 namespace {
-enum HighlighterState { kInitialState, kOpenCommentState, kOpenStringState };
-
 const QColor kGrey{0x80, 0x80, 0x80};
 const QColor kBlue{0x61, 0x96, 0xcc};
 const QColor kYellow{0xa0, 0xa0, 0x33};
 const QColor kYellowOrange{0xff, 0xcc, 0x66};
 const QColor kOrange{0xcc, 0x66, 0x33};
 const QColor kOlive{0x80, 0x80, 0x00};
 const QColor kGreen{0x66, 0x99, 0x66};
-const QColor kViolet{0xcc, 0x99, 0xcd};
+const QColor kViolet{0x99, 0x66, 0x99};
+const QColor kGreyViolet{0xcc, 0xaa, 0xcc};
 
+enum CppHighlighterState { kInitialState, kOpenCommentState, kOpenStringState };
+namespace CppRegex {
 const char* const kNumberRegex =
-    "(?:\\b0x(?:[\\da-f]+(?:\\.[\\da-f]*)?|\\.[\\da-f]+)(?:p[+-]?\\d+)?|(?:\\b\\d+(?:\\.\\d*)?|"
-    "\\B\\.\\d+)(?:e[+-]?\\d+)?)[ful]{0,4}";
+    "(?:\\b0b[01']+|\\b0x(?:[\\da-f']+(?:\\.[\\da-f']*)?|\\.[\\da-f']+)(?:p[+-]?[\\d']+)?|(?:\\b["
+    "\\d']+(?:\\.[\\d']*)?|\\B\\.[\\d']+)(?:e[+-]?[\\d']+)?)[ful]{0,4}";
 const char* const kConstantRegex =
     "__FILE__|__LINE__|__DATE__|__TIME__|__TIMESTAMP__|__func__|EOF|NULL|SEEK_CUR|SEEK_END|SEEK_"
     "SET|stdin|stdout|stderr";
 const char* const kKeywordRegex =
-    "\\b(?:__attribute__|_Alignas|_Alignof|_Atomic|_Bool|_Complex|_Generic|_Imaginary|_Noreturn|_"
-    "Static_assert|_Thread_local|asm|typeof|inline|auto|break|case|char|const|continue|default|do|"
-    "double|else|enum|extern|float|for|goto|if|int|long|register|return|short|signed|sizeof|static|"
-    "struct|switch|typedef|union|unsigned|void|volatile|while|)\\b";
-// Match with comments starting and ending in the same line
+    "\\b(?:alignas|alignof|asm|auto|bool|break|case|catch|char|char8_t|char16_t|char32_t|class|"
+    "compl|concept|const|consteval|constexpr|constinit|const_cast|continue|co_await|co_return|co_"
+    "yield|decltype|default|delete|do|double|dynamic_cast|else|enum|explicit|export|extern|false|"
+    "final|float|for|friend|goto|if|inline|int|int8_t|int16_t|int32_t|int64_t|uint8_t|uint16_t|"
+    "uint32_t|uint64_t|long|mutable|namespace|new|noexcept|nullptr|operator|override|private|"
+    "protected|public|register|reinterpret_cast|requires|return|short|signed|sizeof|static|static_"
+    "assert|static_cast|struct|switch|template|this|thread_local|throw|true|try|typedef|typeid|"
+    "typename|union|unsigned|using|virtual|void|volatile|wchar_t|while)\\b";
```

Comment:
Nice, we already support `concept` and `requires`. You could also add `import` and `export`.